migrations/versions/4663a7a0108b_.py

```diff
@@ -0,0 +1,53 @@
+"""empty message
+
+Revision ID: 4663a7a0108b
+Revises: e83e2897dd6c
+Create Date: 2021-09-14 17:35:33.775558
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '4663a7a0108b'
+down_revision = 'e83e2897dd6c'
+
+from alembic import op
+import sqlalchemy as sa
+import sqlalchemy_utils
+
+import app
+import app.extensions
+
+
+
+def upgrade():
+    """
+    Upgrade Semantic Description:
+        ENTER DESCRIPTION HERE
+    """
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('collaboration', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('initiator_guid', app.extensions.GUID(), nullable=False))
```

Comment:
i realize this upgrade is susceptible to same failure that adding non-nullable `email` to `audit_log` had: namely, any existing _Collaborations_ will prevent this schema change from happening, as `initiator_guid` will be null.  ugh, a tougher upgrade.
i guess in a perfect world, i would write some code here to iterate through `collaboration_user_associations`, find the _initiator users_ there, and *set their guid* on `initiator_guid` here.  :disappointed:

Reply:
ok, think i have some hackery in the migration upgrade that does this now.   :crossed_fingers: